install.sh

```diff
@@ -1,73 +1,294 @@
-    echo "zsh not found. Please install and then re-run installation scripts"
-    exit 1
-elif ! [[ $SHELL =~ .*zsh.* ]]; then
-    echo "Configuring zsh as default shell"
-    chsh -s "$(command -v zsh)"
-fi
-
-# Change the default shell to zsh
-zsh_path="$( command -v zsh )"
-if ! grep "$zsh_path" /etc/shells; then
-    echo "adding $zsh_path to /etc/shells"
-    echo "$zsh_path" | sudo tee -a /etc/shells
-fi
-
-if [[ "$SHELL" != "$zsh_path" ]]; then
-    chsh -s "$zsh_path"
-    echo "default shell changed to $zsh_path"
-fi
-
-echo "Done. Reload your terminal."
+    info "adding xterm-256color-italic.terminfo"
+    tic -x "$DOTFILES/resources/xterm-256color-italic.terminfo"
+}
+
+macos() {
+    if [[ "$(uname)" == "Darwin" ]]; then
+        title "Configuring macOS"
+
+        echo "Finder: show all filename extensions"
+        defaults write NSGlobalDomain AppleShowAllExtensions -bool true
+
+        echo "show hidden files by default"
+        defaults write com.apple.Finder AppleShowAllFiles -bool false
+
+        echo "only use UTF-8 in Terminal.app"
+        defaults write com.apple.terminal StringEncodings -array 4
+
+        echo "expand save dialog by default"
+        defaults write NSGlobalDomain NSNavPanelExpandedStateForSaveMode -bool true
+
+        echo "show the ~/Library folder in Finder"
+        chflags nohidden ~/Library
+
+        echo "Enable full keyboard access for all controls (e.g. enable Tab in modal dialogs)"
+        defaults write NSGlobalDomain AppleKeyboardUIMode -int 3
+
+        echo "Enable subpixel font rendering on non-Apple LCDs"
+        defaults write NSGlobalDomain AppleFontSmoothing -int 2
+
+        echo "Use current directory as default search scope in Finder"
+        defaults write com.apple.finder FXDefaultSearchScope -string "SCcf"
+
+        echo "Show Path bar in Finder"
+        defaults write com.apple.finder ShowPathbar -bool true
+
+        echo "Show Status bar in Finder"
+        defaults write com.apple.finder ShowStatusBar -bool true
+
+        echo "Disable press-and-hold for keys in favor of key repeat"
+        defaults write NSGlobalDomain ApplePressAndHoldEnabled -bool false
+
+        echo "Set a blazingly fast keyboard repeat rate"
+        defaults write NSGlobalDomain KeyRepeat -int 1
+
+        echo "Set a shorter Delay until key repeat"
+        defaults write NSGlobalDomain InitialKeyRepeat -int 15
+
+        echo "Enable tap to click (Trackpad)"
+        defaults write com.apple.driver.AppleBluetoothMultitouch.trackpad Clicking -bool true
+
+        echo "Enable Safari’s debug menu"
+        defaults write com.apple.Safari IncludeInternalDebugMenu -bool true
+
+        echo "Kill affected applications"
+
+        for app in Safari Finder Dock Mail SystemUIServer; do killall "$app" >/dev/null 2>&1; done
+    fi
+}
+
+case "$1" in
+    backup)
+        backup
+        ;;
+    link)
+        link
+        ;;
+    git)
+        setup_git
+        ;;
+    brew)
```

Comment:
Should this be something else as it seems confusing with the actual brew command to install things? I notice on line 282 for all you call the homebrew function and now this

Reply:
Good call. I normalized the names to be prefixed with `setup_` and changed the name of this expected argument from `brew` to `homebrew` to give it a little more distinction from the command.